Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,399 @@
+
+        /// <summary>Get the Cinemachine Pipeline stage that this component implements.
+        /// Always returns the Body stage</summary>
+        public override CinemachineCore.Stage Stage { get { return CinemachineCore.Stage.Body; } }
+
+        /// <summary>
+        /// Report maximum damping time needed for this component.
+        /// </summary>
+        /// <returns>Highest damping setting in this component</returns>
+        public override float GetMaxDampTime() 
+        { 
+            var d2 = AngularDamping;
+            var a = Mathf.Max(m_Damping.x, Mathf.Max(m_Damping.y, m_Damping.z)); 
+            var b = Mathf.Max(d2.x, Mathf.Max(d2.y, d2.z)); 
+            return Mathf.Max(a, b); 
+        }
+        
+        /// <summary>
+        /// Subscribe to onSplineChanged if you'd like to react to changes to the Spline attached to this vcam.
+        /// This action is invoked by the Spline's changed event when a spline property is modified.
+        /// </summary>
+        public event Action onSplineChanged;
+
+        bool m_Registered = false;
+        SplineContainer m_TrackCache;
+        void OnValidate()
+        {
+            if (m_TrackCache != null)
+            {
+                m_TrackCache.Spline.changed -= onSplineChanged;
+                m_TrackCache = m_Track;
+                m_Registered = false;
+            }
+            if (!m_Registered && m_Track != null && m_Track.Spline != null)
+            {
+                m_Registered = true;
+                m_TrackCache = m_Track;
+                m_Track.Spline.changed += onSplineChanged;
+            }
+
+            m_Damping.x = Mathf.Clamp(m_Damping.x, 0, 20);
+            m_Damping.y = Mathf.Clamp(m_Damping.y, 0, 20);
+            m_Damping.z = Mathf.Clamp(m_Damping.z, 0, 20);
+            m_RotationDamping.x = Mathf.Clamp(m_RotationDamping.x, 0, 20);
+            m_RotationDamping.y = Mathf.Clamp(m_RotationDamping.y, 0, 20);
+            m_RotationDamping.z = Mathf.Clamp(m_RotationDamping.z, 0, 20);
+        }
+
+        void UpdateSplineRollOverrideCache()
+        {
+            m_RollOverrideExtension = null;
+            // if vcam has an override, use that
+            if (transform.parent.TryGetComponent(out m_RollOverrideExtension))
+            {
+#if UNITY_EDITOR
+                m_RollOverrideExtension.splineContainer = m_Track; // this is needed by the RollHandle to work in the scene view
+#endif
+            }
+            // else if the spline has an override, use that
+            else if (m_Track.TryGetComponent(out m_RollOverrideExtension)) {}
+        }
+
+        CinemachineSplineRollOverrideExtension m_RollOverrideExtension; // don't use this directly
+        CinemachineSplineRollOverrideExtension RollOverrideExtension
+        {
+            get
+            {
+                if(m_RollOverrideExtension == null)
+                    UpdateSplineRollOverrideCache();
```

Comment:
This doesn't feel very robust.  What if user adds a rollover extension then replaces it with another?

Reply:
It works, because I get notified from by the rollOverrideExtension.

I need to rethink this though 👍